File: ch02-singleton/ex05-interface-based/src/test/java/com/example/CacheTest.java
package com.example;

import static org.junit.jupiter.api.Assertions.assertEquals;

import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.BeforeEach;

class CacheTest {

    private ICache target;

    @BeforeEach
    void beforeEach() {
        target = Cache.getInstance();
    }

    @Test
    void check_object_is_same() {
        ICache cache = Cache.getInstance();
        assertEquals(target, cache);
    }

    @Test
    void check_key_returned() {
        target.setValue("key", "value1");
        String value = target.getValue("key");
        assertEquals("value1", value);
    }

}

File: C#/ch02-singleton/ex01-eager-initialization/src/Main.Tests/CacheTest.cs
using System;
using Xunit;
using Xunit.Sdk;

namespace Main.Tests
{
    public class CacheTest
    {
        private Cache target;
        
        public CacheTest()
        {
            target = Cache.GetInstance();
        }

        [Fact]
        public void Check_Object_IsSame()
        {
            Cache cache = Cache.GetInstance();
            Assert.Equal(target, cache);
        }
    }
}
